use-cases/okta.md

```diff
@@ -0,0 +1,174 @@
+# Use Okta  as OAuth 2.0 server
+
+You are going to test below OAuth 2.0 flow:
+1. Access management ui via a browser :ballot_box_with_check:
+
+## Prerequisites to follow this guide
+
+- Have an account in https://www.okta.com.
+- Docker
+
+## Create your app integration in Okta UI
+
+When using **Okta  as OAuth 2.0 server**, your client app (in our case RabbitMQ) needs a way to trust the security tokens issued to it by the **Okta OIDC Sign-In Widget**. 
+
+A token-based OAuth 2.0 authentication for Single Sign-On (SSO) through API endpoints. 
+
+The first step in establishing that trust is by **creating your app** with the identity platform in OKta.
+
+> :blue_book: More details about App registration in Okta  are available [here](https://help.okta.com/en-us/Content/Topics/Apps/Apps_App_Integration_Wizard_OIDC.htm)
+
+Once you have logged onto your account in [Okta](https://www.okta.com), follow belw steps:
+
+1- In the Admin Console, go to Applications.
+2- Click Create App Integration.
+3- To create an OIDC app integration, select OIDC - OpenID Connect as the Sign-in method.
+4- Choose the type of application to integrate with Okta. Select Web Application, Single-Page Application, or Native Application. In our use case it is Single-Page Application(SPA).
+5- Click Next.
+
+The App Integration Wizard for OIDC has three sections:
+
+1- In General Settings:
+  - **Name**: App integration name: Specify a name for your app integrationn (ex: *rabbitmq-oauth2*)
+  - **Grant type**: select **Authorization Code**  and **Refresh Token**
+  - **Redirect URI**:
+    - On the **Sign-in redirect URIs**  type `http://localhost:15672/js/oidc-oauth/login-callback.html`
+    - Configure the **Sign-out redirect URIs** to: `https://localhost:15672`
+
+2- In Trusted Origins (for Web and Native app integrations):
+
+keep the default values
+
+3- In Assignments: 
+
+Choose **Allow everyone in your organization to access**
+
+Click on **Save**.
+
+
+Note the following values, as you will need it later to configure the `rabbitmq_auth_backend_oauth2` on RabbitMQ side:
+- ClientID.
+
+
+## Create a secret for your app
```

Comment:
I do not think this step is necessary because single page apps are not secure therefore they cannot safely store any credentials besides the client_id. That is why the authorization code with PKCE flow was created.
Nevertheless, I could not find how to create a secret. 

A secret would be necessary for Client Credentials flow. But I could not find anywhere how to enable this flow for non-browser based applications.

Reply:
Sorry for the delay @MarcialRosales, I missed the notification.
Changes are done

Reply:
Hello @MarcialRosales , changes are done as asked, can you please review them ?
thanks

Reply:
It is perfect @seifrajhi . I am merging it !! Thanks a lot for contributing it !

Reply:
I will try to add token generation script also from Okta side for curl command 
The existing script does not cover that